backend/layers/processing/upload_failures/app.py

```diff
@@ -104,6 +114,11 @@ def get_failure_slack_notification_message(
     batch_url = aws_batch_job_url_fmt_str.format(aws_region=aws_region, job_id=job_id)
     step_function_url = aws_sfn_url_fmt_str.format(aws_region=aws_region, execution_arn=execution_arn)
     collection_version_url = f"https://cellxgene.cziscience.com/collections/{collection_version_id}"
+
+    # Generate canonical collection URL
+    version = get_business_logic().get_collection_version(CollectionVersionId(collection_version_id))
+    collection_url = f"https://cellxgene.cziscience.com/collections/{version.collection_id.id}"
```

Comment:
should this be using the new logic?
```suggestion
    collection_url = self.business_logic.get_collection_url(version.collection_id.id)
```

Reply:
I had considered that, but this line which generates the collection version url: https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/layers/processing/upload_failures/app.py#L106 uses the hard-coded prod URL. My understanding is that this is only run in prod which is why that's okay, but I'm not 100% sure.

I'm fine with updating it to use `self.business_logic.get_collection_url` but I think we'd want to update both the version URL and the canonical URL.

Reply:
It's fine as is for now.